### ISSUE TITLE
Accessibility failure: Button should not be used in Upload

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#46215]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/upload-by-clicking-antd-5-11-5-forked-tcj9py)

### Steps to reproduce

Render the `Upload` component using code from the Storybook examples where `Button` is a child. eg:
```
  <Upload {...props}>
    <Button icon={<UploadOutlined />}>Click to Upload</Button>
  </Upload>
```

### What is expected?

- No accessibility failures.
- Correct focus treatment.

### What is actually happening?

- [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) failure: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies
- Incorrect focus treatment: Initially tabbing into the component (which focuses the top-level button) does not result in the visual focus treatment appearing. It's only after tabbing again (which focuses the inner button) that the visual focus treatment appears correctly.

The root cause of this issue is that the `Upload` component renders an element with `role="button"` and the `Button` component renders a `<button>` element. This results in an interactive element being inside another interactive element which is an a11y failure.

```
<span class="ant-upload-wrapper css-2q8sxy">
  <div class="ant-upload ant-upload-select">
    <span tabindex="0" class="ant-upload" role="button">
      <input type="file" accept="" style="display: none;">
      <button type="button" class="ant-btn css-2q8sxy ant-btn-default">
        <span class="ant-btn-icon">
          <span role="img" aria-label="upload" class="anticon anticon-upload">
            <svg viewBox="64 64 896 896" focusable="false" data-icon="upload" width="1em" height="1em" fill="currentColor" aria-hidden="true">
              <path d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"></path>
            </svg>
          </span>
        </span>
        <span>Click to Upload</span>
      </button>
    </span>
  </div>
  <div class="ant-upload-list ant-upload-list-text"></div>
</span>
```

| Environment | Info |
| --- | --- |
| antd | 5.11.5 |
| React | 18 |
| System | Windows |
| Browser | Chrome |

---
It's my belief that the correct approach is for there to be an expectation that `Upload` will render a button element and the children of `Upload` should, therefore, not contain any interactive elements (eg. buttons, anchors). The implications of this would be:
- `Upload` needs to have the correct (button) CSS treatments applied for focus state, etc.
- Storybook examples for `Upload` should be updated to not use interactive elements as children.

The application of new CSS for `Upload` would cause visual issues for consumers who would upgrade to the version with this change that were still using `Button` as children so perhaps this would need to be done in a major release.

**What do the maintainers think?**

**Comment by mellis481** [comment]
@afc163 To confirm, the PR you would accept would contain:
- CSS updates to `Upload`.
- Removal of interactive elements as children of `Upload` in Storybook examples.]

Correct?

**Comment by mellis481** [comment]
Yes, removing the `role="button"` from `.ant-upload` is probably a better approach. We'd also need to remove the `tabindex="0"` as well.

It looks like these changes would need to be made in rc-upload since the component in that repo (which the Ant `Upload` component wraps) does not allow you to override role or tabindex. I'll create a PR there.

**Comment by mellis481** [comment]
@afc163 Would your preference be that the current behavior (`role="button"` and `tabindex="0"` set on the upload element) to be retained as the default, but allow for consumers to override it OR do you want those attributes to be removed and unable to be set by consumers? (This is so we can prepare the correct rc-upload PR)

**Comment by afc163** [comment]
Yes, LGTM.

**Comment by mellis481** [comment]
> Yes, LGTM.

@afc163 Can you please clarify? My last question was which of the two options you prefer.

**Comment by vtx-anton-chashchin** [comment]
@afc163 Based on the discussion above, could you check the PR please?
https://github.com/react-component/upload/pull/521

**Comment by afc163** [comment]
> the current behavior (role="button" and tabindex="0" set on the upload element) to be retained as the default, but allow for consumers to override it

I prefer this one.
